src/derivative_operators/BC_operators.jl

```diff
@@ -139,14 +145,150 @@ function LinearAlgebra.Array(Q::BoundaryPaddedVector)
+            for i in 1:size[1]
+                @views out[i,:,j] = A*u[i,:,j]
+            end
+        end
+    elseif dim == 3
+        for j in 1:size[2]
+            for i in 1:size[1]
+                @views out[i,j,:] = A*u[i,j,:]
+            end
+        end
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return out
+end
+
+
+# The BC is applied stripwise and the boundary Arrays built from the l/r of the BoundaryPaddedVectors
+@inline function slicemul(A::SingleLayerBC, u::AbstractArray{T, 2}, dim::Integer) where T
+    s = size(u)
+    if dim == 1
+        lower = zeros(T, s[2])
+        upper = deepcopy(lower)
+        for i in 1:s[2]
+            tmp = A*u[:,i]
+            lower[i] = tmp.l
+            upper[i] = tmp.r
+        end
+    elseif dim == 2
+        lower = zeros(T, s[1])
+        upper = deepcopy(lower)
+        for i in 1:s[1]
+            tmp = A*u[i,:]
+            lower[i] = tmp.l
+            upper[i] = tmp.r
+        end
+    elseif dim == 3
+        throw("The 3 dimensional Method should be being called, not this one. Check dispatch.")
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return lower, upper
+end
+
+@inline function slicemul(A::SingleLayerBC, u::AbstractArray{T, 3}, dim::Integer) where T
+    s = Array(size(u))
+    if dim == 1
+        lower = zeros(T, s[2], s[3])
+        upper = deepcopy(lower)
+        for j in 1:s[3]
+            for i in 1:s[2]
+                tmp = A*u[:,i,j]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    elseif dim == 2
+        lower = zeros(T, s[1], s[3])
+        upper = deepcopy(lower)
+        for j in 1:s[3]
+            for i in 1:s[1]
+                tmp = A*u[i,:,j]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    elseif dim == 3
+        lower = zeros(T, s[1], s[2])
+        upper = deepcopy(lower)
+        for j in 1:s[2]
+            for i in 1:s[1]
+                tmp = A*u[i,j,:]
+                lower[i,j] = tmp.l
+                upper[i,j] = tmp.r
+            end
+        end
+    else
+        throw("Dim greater than 3 not supported!")
+    end
+    return lower, upper
+end
+
+struct MultiDimensionalSingleLayerBC{T, N} <: MultiDimensionalBC{T, N}
+    BC::Vector{SingleLayerBC{T}} # I think this has to be an array of non concrete BCs to allow different BCs on different dims
+    MultiDimBC(BCs::AbstractBC{T}...) where T = new{T, length(BCs)}(Array(BCs))
+    MultiDimBC(BCs::AbstractVector{AbstractBC{T}}) where T = new{T, length(BCs)}(Array(BCs))
+end
+
+
+"""
+Higher dimensional generalization of BoundaryPaddedVector, pads an array of dimension N with 2*N arrays of dimension N-1, stored in lower and upper.
+
+"""
+struct BoundaryPaddedArray{T, N, V <: AbstractArray{T}, B <: AbstractArray{T}} <: AbstractBoundaryPaddedArray{T, N}
+    lower::Vector{B}
+    upper::Vector{B}
```

Comment:
for this, don't you need left right top bottom?

Reply:
Actually, could this get a similar N-dimensional interpretation as MultiDimensionalSingleLayerBC?

Reply:
This is N dimensional, lower is a vector of BCs to be applied to the lower index end, upper to the higher index end. Doesn't allow for mixed BCs along a boundary yet though